src/main/java/dog/pawbook/model/dog/DateOfBirth.java

```diff
@@ -0,0 +1,75 @@
+package dog.pawbook.model.dog;
+
+import static dog.pawbook.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a Dog's breed in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBreed(String)}
+ */
+public class DateOfBirth {
+
+    public static final String MESSAGE_CONSTRAINTS = "Age should only contain numbers and should not be blank.";
+    public static final String VALIDATION_REGEX = "^(3[01]|[12][0-9]|0[1-9])-(1[0-2]|0[1-9])-[0-9]{4}$";
+
+    public final String value;
```

Comment:
Is it better to store it as `LocalDate` instead? this will simplify a lot of things especially for the getAge method. And it will be easier for us to manipulate into any other format elsewhere, say the UI and json.